src/modules/window/sdl/Window.cpp

```diff
@@ -1443,6 +1443,33 @@ bool Window::isMinimized() const
 	return window != nullptr && (SDL_GetWindowFlags(window) & SDL_WINDOW_MINIMIZED);
 }
 
+static void get_displaymode(SDL_DisplayMode* mode, SDL_Window* window)
```

Comment:
Usually I prefer to keep the #if branches in code where it's used instead of splitting it into its own function, since then we can just delete the other branch once we move to SDL3-only, without having to refactor anything else.

(This also applies to the swapinterval function.)

Reply:
And how about writing the code for sdl3, and add a backwards compat function with this same prototype for sdl2?
Then only the function has to be removed when dropping sdl2 support

Reply:
I don't think this code really benefits from a function as-is, since you'd have to navigate away from where it's used to see what it does but it doesn't do something that's especially useful outside of that context.